app/lib/pre_assembly/batch.rb

```diff
@@ -71,20 +71,22 @@ def object_filenames_unique?(dobj)
     # For each container, creates a new Digitalobject.
     # @return [Array<DigitalObject>]
     # rubocop:disable Metrics/AbcSize
-    def digital_objects
-      @digital_objects ||= discover_containers_via_manifest.each_with_index.map do |c, i|
+    def digital_objects(&_block)
+      return enum_for(:digital_objects) unless block_given?
+
+      discover_containers_via_manifest.each_with_index.map do |c, i|
```

Comment:
could you change "c" to "container" or something like that?

Reply:
done